# 최적화된 AWQ 양자화

## 왜 AWQ인가?

### 벤치마크 결과 ([vLLM Quantization Guide](https://docs.jarvislabs.ai/blog/vllm-quantization-complete-guide-benchmarks))

| 방식 | 처리량 (tok/s) | TTFT | ITL |
|------|---------------|------|-----|
| Baseline | 461 | 151ms | 21.2ms |
| GPTQ | 276 | 165ms | 35.5ms |
| **Marlin-GPTQ** | 712 | 118ms | 13.8ms |
| **Marlin-AWQ** | **741** | 137ms | **12.6ms** |

### 핵심 발견
1. **Marlin-AWQ가 가장 빠름** (741 tok/s)
2. AWQ는 "활성화 인식" 양자화 → 중요한 가중치 보존
3. vLLM이 Marlin 커널 자동 적용

---

# 1. Import

In [1]:
import os
import torch
import shutil
from pathlib import Path

from datasets import load_dataset
from transformers import AutoModelForCausalLM, AutoTokenizer

from llmcompressor import oneshot
from llmcompressor.modifiers.quantization import GPTQModifier

print(f"PyTorch: {torch.__version__}")
print(f"CUDA: {torch.cuda.is_available()}")
if torch.cuda.is_available():
    print(f"GPU: {torch.cuda.get_device_name(0)}")
else:
    print("⚠️ CPU 모드로 실행됩니다")
print("\n✅ Import 완료!")

PyTorch: 2.9.1
CUDA: False
⚠️ CPU 모드로 실행됩니다

✅ Import 완료!


# 2. 설정

### AWQ vs GPTQ 차이점

| 항목 | GPTQ | AWQ |
|------|------|-----|
| 핵심 | Hessian 기반 | 활성화 기반 |
| 속도 | 빠름 | 더 빠름 |
| actorder | 필요 | **"static" 권장** |

In [2]:
# ============================================================================
# 모델 설정
# ============================================================================
MODEL_ID = "./open/base_model"
OUT_DIR = "./model"

# ============================================================================
# 데이터셋 설정
# ============================================================================
DATASET_ID = "LGAI-EXAONE/MANTA-1M"
DATASET_SPLIT = "train"

# ============================================================================
# 캘리브레이션 설정 (베이스라인과 동일)
# ============================================================================
NUM_CALIBRATION_SAMPLES = 256
MAX_SEQUENCE_LENGTH = 512

# ============================================================================
# AWQ 스타일 양자화 설정
# ============================================================================
# llmcompressor는 GPTQModifier로 AWQ 스타일 적용
# actorder="static"이 AWQ와 유사한 효과

SCHEME = "W4A16"
TARGETS = ["Linear"]
IGNORE = ["embed_tokens", "lm_head"]

# AWQ 최적화 설정
BLOCK_SIZE = 128         # Marlin 호환
DAMPENING_FRAC = 0.01    # AWQ 스타일 (0.01 권장)
ACTORDER = "static"      # AWQ 스타일 (static 권장)

ORIGINAL_MODEL_SIZE_GB = 2.56

print("=" * 60)
print("AWQ 스타일 양자화 설정")
print("=" * 60)
print(f"MODEL_ID: {MODEL_ID}")
print(f"SCHEME: {SCHEME}")
print(f"SAMPLES: {NUM_CALIBRATION_SAMPLES}")
print("---")
print("⭐ AWQ 최적화:")
print(f"  BLOCK_SIZE: {BLOCK_SIZE}")
print(f"  DAMPENING_FRAC: {DAMPENING_FRAC}")
print(f"  ACTORDER: {ACTORDER}")
print("=" * 60)

AWQ 스타일 양자화 설정
MODEL_ID: ./open/base_model
SCHEME: W4A16
SAMPLES: 256
---
⭐ AWQ 최적화:
  BLOCK_SIZE: 128
  DAMPENING_FRAC: 0.01
  ACTORDER: static


# 3. 모델 로드

In [3]:
print("[INFO] 모델 로드 중...")

tokenizer = AutoTokenizer.from_pretrained(
    MODEL_ID,
    trust_remote_code=True,
)

if torch.cuda.is_available():
    model = AutoModelForCausalLM.from_pretrained(
        MODEL_ID,
        torch_dtype=torch.bfloat16,
        trust_remote_code=True,
    )
else:
    model = AutoModelForCausalLM.from_pretrained(
        MODEL_ID,
        torch_dtype=torch.float32,
        trust_remote_code=True,
        device_map="cpu",
    )

print(f"[INFO] 모델 파라미터: {model.num_parameters():,}")
print("[INFO] 모델 로드 완료")

[INFO] 모델 로드 중...


`torch_dtype` is deprecated! Use `dtype` instead!


[INFO] 모델 파라미터: 1,279,391,488
[INFO] 모델 로드 완료


# 4. 데이터셋 로드

In [4]:
print("[INFO] 캘리브레이션 데이터 로드 중...")

ds = load_dataset(
    DATASET_ID,
    split=f"{DATASET_SPLIT}[:{NUM_CALIBRATION_SAMPLES}]",
)

def preprocess(example):
    return {
        "text": tokenizer.apply_chat_template(
            example["conversations"],
            add_generation_prompt=True,
            tokenize=False
        )
    }

ds = ds.map(preprocess)

print(f"[INFO] 데이터셋 크기: {len(ds)}")

[INFO] 캘리브레이션 데이터 로드 중...
[INFO] 데이터셋 크기: 256


# 5. AWQ 스타일 양자화

In [5]:
print("[INFO] AWQ 스타일 양자화 시작")
print(f"  - scheme: {SCHEME}")
print(f"  - samples: {NUM_CALIBRATION_SAMPLES}")
print(f"  - actorder: {ACTORDER} (AWQ 스타일)")
print(f"  - dampening: {DAMPENING_FRAC}")

if torch.cuda.is_available():
    print("\n🚀 GPU 모드: 10-20분 예상\n")
else:
    print("\n⏳ CPU 모드: 15-30분 예상\n")

recipe = [
    GPTQModifier(
        scheme=SCHEME,
        targets=TARGETS,
        ignore=IGNORE,
        block_size=BLOCK_SIZE,
        dampening_frac=DAMPENING_FRAC,
        actorder=ACTORDER,  # static = AWQ 스타일
    )
]

oneshot(
    model=model,
    dataset=ds,
    recipe=recipe,
    max_seq_length=MAX_SEQUENCE_LENGTH,
    num_calibration_samples=NUM_CALIBRATION_SAMPLES,
)

print("\n[INFO] 양자화 완료!")

[INFO] AWQ 스타일 양자화 시작
  - scheme: W4A16
  - samples: 256
  - actorder: static (AWQ 스타일)
  - dampening: 0.01

⏳ CPU 모드: 15-30분 예상



Tokenizing:   0%|          | 0/256 [00:00<?, ? examples/s]

2026-02-10T15:24:42.035217+0900 | reset | INFO - Compression lifecycle reset
2026-02-10T15:24:42.037084+0900 | from_modifiers | INFO - Creating recipe from modifiers
2026-02-10T15:24:42.063552+0900 | initialize | INFO - Compression lifecycle initialized for 1 modifiers
2026-02-10T15:24:42.064212+0900 | IndependentPipeline | INFO - Inferred `SequentialPipeline` for `GPTQModifier`
2026-02-10T15:24:42.070899+0900 | dispatch_for_sequential | WARNING - CUDA/XPU is not available! Compressing model on CPU instead


W0210 15:24:42.108000 95207 torch/fx/_symbolic_trace.py:52] is_fx_tracing will return true for both fx.symbolic_trace and torch.export. Please use is_fx_tracing_symbolic_tracing() for specifically fx.symbolic_trace or torch.compiler.is_compiling() for specifically torch.export/compile.
(1/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:57<00:00,  4.46it/s]

2026-02-10T15:25:39.788952+0900 | compress_modules | INFO - Quantizing model.layers.0.self_attn.q_proj using 256 samples


2026-02-10T15:25:40.268824+0900 | compress | METRIC - time 0.48s
2026-02-10T15:25:40.269307+0900 | compress | METRIC - error 1.12
2026-02-10T15:25:40.272112+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:25:40.272563+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:25:40.275239+0900 | compress_modules | INFO - Quantizing model.layers.0.self_attn.k_proj using 256 samples
2026-02-10T15:25:40.511483+0900 | compress | METRIC - time 0.24s
2026-02-10T15:25:40.512012+0900 | compress | METRIC - error 0.33
2026-02-10T15:25:40.513170+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:25:40.513454+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:25:40.514238+0900 | compress_modules | INFO - Quantizing model.layers.0.self_attn.v_proj using 256 samples
2026-02-10T15:25:40.759394+0900 | compress | METRIC - time 0.24s
2026-02-10T15:25:40.75

(2/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:58<00:00,  4.38it/s]

2026-02-10T15:26:53.400326+0900 | compress_modules | INFO - Quantizing model.layers.1.self_attn.q_proj using 256 samples


2026-02-10T15:26:53.831262+0900 | compress | METRIC - time 0.43s
2026-02-10T15:26:53.831898+0900 | compress | METRIC - error 4.77
2026-02-10T15:26:53.834062+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:26:53.834517+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:26:53.836904+0900 | compress_modules | INFO - Quantizing model.layers.1.self_attn.k_proj using 256 samples
2026-02-10T15:26:54.089118+0900 | compress | METRIC - time 0.25s
2026-02-10T15:26:54.089621+0900 | compress | METRIC - error 1.36
2026-02-10T15:26:54.090760+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:26:54.091061+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:26:54.091866+0900 | compress_modules | INFO - Quantizing model.layers.1.self_attn.v_proj using 256 samples
2026-02-10T15:26:54.338334+0900 | compress | METRIC - time 0.25s
2026-02-10T15:26:54.33

(3/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:56<00:00,  4.49it/s]

2026-02-10T15:28:04.528181+0900 | compress_modules | INFO - Quantizing model.layers.2.self_attn.q_proj using 256 samples


2026-02-10T15:28:04.942674+0900 | compress | METRIC - time 0.41s
2026-02-10T15:28:04.943187+0900 | compress | METRIC - error 12.96
2026-02-10T15:28:04.944461+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:28:04.944772+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:28:04.946966+0900 | compress_modules | INFO - Quantizing model.layers.2.self_attn.k_proj using 256 samples
2026-02-10T15:28:05.197678+0900 | compress | METRIC - time 0.25s
2026-02-10T15:28:05.198196+0900 | compress | METRIC - error 3.64
2026-02-10T15:28:05.199398+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:28:05.199673+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:28:05.200607+0900 | compress_modules | INFO - Quantizing model.layers.2.self_attn.v_proj using 256 samples
2026-02-10T15:28:05.457030+0900 | compress | METRIC - time 0.26s
2026-02-10T15:28:05.4

(4/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:58<00:00,  4.34it/s]

2026-02-10T15:29:17.419963+0900 | compress_modules | INFO - Quantizing model.layers.3.self_attn.q_proj using 256 samples


2026-02-10T15:29:17.834639+0900 | compress | METRIC - time 0.41s
2026-02-10T15:29:17.835142+0900 | compress | METRIC - error 26.44
2026-02-10T15:29:17.837710+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:29:17.838154+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:29:17.839984+0900 | compress_modules | INFO - Quantizing model.layers.3.self_attn.k_proj using 256 samples
2026-02-10T15:29:18.074677+0900 | compress | METRIC - time 0.23s
2026-02-10T15:29:18.075130+0900 | compress | METRIC - error 7.47
2026-02-10T15:29:18.076084+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:29:18.076343+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:29:18.077127+0900 | compress_modules | INFO - Quantizing model.layers.3.self_attn.v_proj using 256 samples
2026-02-10T15:29:18.334249+0900 | compress | METRIC - time 0.26s
2026-02-10T15:29:18.3

(5/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:58<00:00,  4.37it/s]

2026-02-10T15:30:29.605291+0900 | compress_modules | INFO - Quantizing model.layers.4.self_attn.q_proj using 256 samples


2026-02-10T15:30:29.993881+0900 | compress | METRIC - time 0.39s
2026-02-10T15:30:29.994362+0900 | compress | METRIC - error 50.32
2026-02-10T15:30:29.995961+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:30:29.996260+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:30:29.997986+0900 | compress_modules | INFO - Quantizing model.layers.4.self_attn.k_proj using 256 samples
2026-02-10T15:30:30.235480+0900 | compress | METRIC - time 0.24s
2026-02-10T15:30:30.235870+0900 | compress | METRIC - error 13.93
2026-02-10T15:30:30.236841+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:30:30.237091+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:30:30.237944+0900 | compress_modules | INFO - Quantizing model.layers.4.self_attn.v_proj using 256 samples
2026-02-10T15:30:30.469129+0900 | compress | METRIC - time 0.23s
2026-02-10T15:30:30.

(6/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:57<00:00,  4.43it/s]

2026-02-10T15:31:41.278308+0900 | compress_modules | INFO - Quantizing model.layers.5.self_attn.q_proj using 256 samples


2026-02-10T15:31:41.718936+0900 | compress | METRIC - time 0.44s
2026-02-10T15:31:41.719463+0900 | compress | METRIC - error 81.40
2026-02-10T15:31:41.722023+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:31:41.722430+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:31:41.724297+0900 | compress_modules | INFO - Quantizing model.layers.5.self_attn.k_proj using 256 samples
2026-02-10T15:31:41.983888+0900 | compress | METRIC - time 0.26s
2026-02-10T15:31:41.984315+0900 | compress | METRIC - error 23.90
2026-02-10T15:31:41.985306+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:31:41.985571+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:31:41.986344+0900 | compress_modules | INFO - Quantizing model.layers.5.self_attn.v_proj using 256 samples
2026-02-10T15:31:42.231415+0900 | compress | METRIC - time 0.24s
2026-02-10T15:31:42.

(7/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:59<00:00,  4.32it/s]

2026-02-10T15:32:54.584419+0900 | compress_modules | INFO - Quantizing model.layers.6.self_attn.q_proj using 256 samples


2026-02-10T15:32:55.004244+0900 | compress | METRIC - time 0.42s
2026-02-10T15:32:55.004706+0900 | compress | METRIC - error 118.11
2026-02-10T15:32:55.006842+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:32:55.007143+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:32:55.009000+0900 | compress_modules | INFO - Quantizing model.layers.6.self_attn.k_proj using 256 samples
2026-02-10T15:32:55.269811+0900 | compress | METRIC - time 0.26s
2026-02-10T15:32:55.270269+0900 | compress | METRIC - error 32.48
2026-02-10T15:32:55.271329+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:32:55.271639+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:32:55.272484+0900 | compress_modules | INFO - Quantizing model.layers.6.self_attn.v_proj using 256 samples
2026-02-10T15:32:55.556740+0900 | compress | METRIC - time 0.28s
2026-02-10T15:32:55

(8/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:00<00:00,  4.23it/s]

2026-02-10T15:34:09.955249+0900 | compress_modules | INFO - Quantizing model.layers.7.self_attn.q_proj using 256 samples


2026-02-10T15:34:10.423336+0900 | compress | METRIC - time 0.47s
2026-02-10T15:34:10.423906+0900 | compress | METRIC - error 178.31
2026-02-10T15:34:10.426392+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:34:10.426751+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:34:10.428859+0900 | compress_modules | INFO - Quantizing model.layers.7.self_attn.k_proj using 256 samples
2026-02-10T15:34:10.702618+0900 | compress | METRIC - time 0.27s
2026-02-10T15:34:10.703665+0900 | compress | METRIC - error 50.15
2026-02-10T15:34:10.704781+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:34:10.705129+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:34:10.706227+0900 | compress_modules | INFO - Quantizing model.layers.7.self_attn.v_proj using 256 samples
2026-02-10T15:34:10.967005+0900 | compress | METRIC - time 0.26s
2026-02-10T15:34:10

(9/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:01<00:00,  4.19it/s]

2026-02-10T15:35:26.163786+0900 | compress_modules | INFO - Quantizing model.layers.8.self_attn.q_proj using 256 samples


2026-02-10T15:35:26.667098+0900 | compress | METRIC - time 0.50s
2026-02-10T15:35:26.667607+0900 | compress | METRIC - error 195.12
2026-02-10T15:35:26.670515+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:35:26.671145+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:35:26.673369+0900 | compress_modules | INFO - Quantizing model.layers.8.self_attn.k_proj using 256 samples
2026-02-10T15:35:27.074526+0900 | compress | METRIC - time 0.40s
2026-02-10T15:35:27.075030+0900 | compress | METRIC - error 55.68
2026-02-10T15:35:27.076145+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:35:27.076507+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:35:27.077758+0900 | compress_modules | INFO - Quantizing model.layers.8.self_attn.v_proj using 256 samples
2026-02-10T15:35:27.384046+0900 | compress | METRIC - time 0.31s
2026-02-10T15:35:27

(10/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:57<00:00,  4.46it/s]

2026-02-10T15:36:41.048060+0900 | compress_modules | INFO - Quantizing model.layers.9.self_attn.q_proj using 256 samples


2026-02-10T15:36:41.495224+0900 | compress | METRIC - time 0.45s
2026-02-10T15:36:41.495769+0900 | compress | METRIC - error 260.15
2026-02-10T15:36:41.498281+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:36:41.498681+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:36:41.500641+0900 | compress_modules | INFO - Quantizing model.layers.9.self_attn.k_proj using 256 samples
2026-02-10T15:36:41.768519+0900 | compress | METRIC - time 0.27s
2026-02-10T15:36:41.769174+0900 | compress | METRIC - error 76.72
2026-02-10T15:36:41.770167+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:36:41.770483+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:36:41.771296+0900 | compress_modules | INFO - Quantizing model.layers.9.self_attn.v_proj using 256 samples
2026-02-10T15:36:42.029545+0900 | compress | METRIC - time 0.26s
2026-02-10T15:36:42

(11/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:00<00:00,  4.25it/s]

2026-02-10T15:37:56.475970+0900 | compress_modules | INFO - Quantizing model.layers.10.self_attn.q_proj using 256 samples


2026-02-10T15:37:56.937063+0900 | compress | METRIC - time 0.46s
2026-02-10T15:37:56.937666+0900 | compress | METRIC - error 282.98
2026-02-10T15:37:56.940633+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:37:56.941122+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:37:56.943177+0900 | compress_modules | INFO - Quantizing model.layers.10.self_attn.k_proj using 256 samples
2026-02-10T15:37:57.203964+0900 | compress | METRIC - time 0.26s
2026-02-10T15:37:57.204604+0900 | compress | METRIC - error 76.06
2026-02-10T15:37:57.205754+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:37:57.206078+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:37:57.206945+0900 | compress_modules | INFO - Quantizing model.layers.10.self_attn.v_proj using 256 samples
2026-02-10T15:37:57.476144+0900 | compress | METRIC - time 0.27s
2026-02-10T15:37:

(12/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:59<00:00,  4.31it/s]

2026-02-10T15:39:11.758473+0900 | compress_modules | INFO - Quantizing model.layers.11.self_attn.q_proj using 256 samples


2026-02-10T15:39:12.217441+0900 | compress | METRIC - time 0.46s
2026-02-10T15:39:12.218069+0900 | compress | METRIC - error 307.88
2026-02-10T15:39:12.219366+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:39:12.219673+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:39:12.221842+0900 | compress_modules | INFO - Quantizing model.layers.11.self_attn.k_proj using 256 samples
2026-02-10T15:39:12.499725+0900 | compress | METRIC - time 0.28s
2026-02-10T15:39:12.500318+0900 | compress | METRIC - error 87.01
2026-02-10T15:39:12.501539+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:39:12.501893+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:39:12.502884+0900 | compress_modules | INFO - Quantizing model.layers.11.self_attn.v_proj using 256 samples
2026-02-10T15:39:12.783488+0900 | compress | METRIC - time 0.28s
2026-02-10T15:39:

(13/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:59<00:00,  4.32it/s]

2026-02-10T15:40:26.597920+0900 | compress_modules | INFO - Quantizing model.layers.12.self_attn.q_proj using 256 samples


2026-02-10T15:40:27.253903+0900 | compress | METRIC - time 0.65s
2026-02-10T15:40:27.255193+0900 | compress | METRIC - error 344.53
2026-02-10T15:40:27.258798+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:40:27.260646+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:40:27.266012+0900 | compress_modules | INFO - Quantizing model.layers.12.self_attn.k_proj using 256 samples
2026-02-10T15:40:27.691524+0900 | compress | METRIC - time 0.42s
2026-02-10T15:40:27.692077+0900 | compress | METRIC - error 94.53
2026-02-10T15:40:27.693255+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:40:27.693571+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:40:27.694549+0900 | compress_modules | INFO - Quantizing model.layers.12.self_attn.v_proj using 256 samples
2026-02-10T15:40:27.954949+0900 | compress | METRIC - time 0.26s
2026-02-10T15:40:

(14/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:00<00:00,  4.21it/s]

2026-02-10T15:41:42.867292+0900 | compress_modules | INFO - Quantizing model.layers.13.self_attn.q_proj using 256 samples


2026-02-10T15:41:43.307840+0900 | compress | METRIC - time 0.44s
2026-02-10T15:41:43.308377+0900 | compress | METRIC - error 386.91
2026-02-10T15:41:43.309756+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:41:43.310113+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:41:43.312096+0900 | compress_modules | INFO - Quantizing model.layers.13.self_attn.k_proj using 256 samples
2026-02-10T15:41:43.578646+0900 | compress | METRIC - time 0.27s
2026-02-10T15:41:43.579155+0900 | compress | METRIC - error 108.32
2026-02-10T15:41:43.580231+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:41:43.580544+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:41:43.581392+0900 | compress_modules | INFO - Quantizing model.layers.13.self_attn.v_proj using 256 samples
2026-02-10T15:41:43.838116+0900 | compress | METRIC - time 0.26s
2026-02-10T15:41

(15/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:59<00:00,  4.30it/s]

2026-02-10T15:42:57.509782+0900 | compress_modules | INFO - Quantizing model.layers.14.self_attn.q_proj using 256 samples


2026-02-10T15:42:57.903748+0900 | compress | METRIC - time 0.39s
2026-02-10T15:42:57.904233+0900 | compress | METRIC - error 419.89
2026-02-10T15:42:57.905346+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:42:57.905639+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:42:57.907560+0900 | compress_modules | INFO - Quantizing model.layers.14.self_attn.k_proj using 256 samples
2026-02-10T15:42:58.157266+0900 | compress | METRIC - time 0.25s
2026-02-10T15:42:58.157741+0900 | compress | METRIC - error 126.15
2026-02-10T15:42:58.159187+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:42:58.159653+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:42:58.160911+0900 | compress_modules | INFO - Quantizing model.layers.14.self_attn.v_proj using 256 samples
2026-02-10T15:42:58.407298+0900 | compress | METRIC - time 0.25s
2026-02-10T15:42

(16/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:59<00:00,  4.31it/s]

2026-02-10T15:44:11.418349+0900 | compress_modules | INFO - Quantizing model.layers.15.self_attn.q_proj using 256 samples


2026-02-10T15:44:11.832404+0900 | compress | METRIC - time 0.41s
2026-02-10T15:44:11.832933+0900 | compress | METRIC - error 435.28
2026-02-10T15:44:11.834061+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:44:11.834378+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:44:11.836766+0900 | compress_modules | INFO - Quantizing model.layers.15.self_attn.k_proj using 256 samples
2026-02-10T15:44:12.102159+0900 | compress | METRIC - time 0.26s
2026-02-10T15:44:12.102610+0900 | compress | METRIC - error 122.69
2026-02-10T15:44:12.103904+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:44:12.104215+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:44:12.105171+0900 | compress_modules | INFO - Quantizing model.layers.15.self_attn.v_proj using 256 samples
2026-02-10T15:44:12.428622+0900 | compress | METRIC - time 0.32s
2026-02-10T15:44

(17/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:59<00:00,  4.32it/s]

2026-02-10T15:45:25.634214+0900 | compress_modules | INFO - Quantizing model.layers.16.self_attn.q_proj using 256 samples


2026-02-10T15:45:26.063310+0900 | compress | METRIC - time 0.43s
2026-02-10T15:45:26.063807+0900 | compress | METRIC - error 515.29
2026-02-10T15:45:26.064892+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:45:26.065216+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:45:26.067000+0900 | compress_modules | INFO - Quantizing model.layers.16.self_attn.k_proj using 256 samples
2026-02-10T15:45:26.329370+0900 | compress | METRIC - time 0.26s
2026-02-10T15:45:26.329936+0900 | compress | METRIC - error 134.84
2026-02-10T15:45:26.331181+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:45:26.331556+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:45:26.332481+0900 | compress_modules | INFO - Quantizing model.layers.16.self_attn.v_proj using 256 samples
2026-02-10T15:45:26.599205+0900 | compress | METRIC - time 0.27s
2026-02-10T15:45

(18/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:59<00:00,  4.28it/s]

2026-02-10T15:46:40.451547+0900 | compress_modules | INFO - Quantizing model.layers.17.self_attn.q_proj using 256 samples


2026-02-10T15:46:40.864628+0900 | compress | METRIC - time 0.41s
2026-02-10T15:46:40.865138+0900 | compress | METRIC - error 532.66
2026-02-10T15:46:40.866267+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:46:40.866536+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:46:40.868182+0900 | compress_modules | INFO - Quantizing model.layers.17.self_attn.k_proj using 256 samples
2026-02-10T15:46:41.130447+0900 | compress | METRIC - time 0.26s
2026-02-10T15:46:41.130955+0900 | compress | METRIC - error 144.69
2026-02-10T15:46:41.132068+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:46:41.132340+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:46:41.133069+0900 | compress_modules | INFO - Quantizing model.layers.17.self_attn.v_proj using 256 samples
2026-02-10T15:46:41.393732+0900 | compress | METRIC - time 0.26s
2026-02-10T15:46

(19/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:56<00:00,  4.53it/s]

2026-02-10T15:47:52.272446+0900 | compress_modules | INFO - Quantizing model.layers.18.self_attn.q_proj using 256 samples


2026-02-10T15:47:52.663779+0900 | compress | METRIC - time 0.39s
2026-02-10T15:47:52.664274+0900 | compress | METRIC - error 585.64
2026-02-10T15:47:52.665454+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:47:52.665727+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:47:52.667392+0900 | compress_modules | INFO - Quantizing model.layers.18.self_attn.k_proj using 256 samples
2026-02-10T15:47:52.908748+0900 | compress | METRIC - time 0.24s
2026-02-10T15:47:52.909145+0900 | compress | METRIC - error 166.38
2026-02-10T15:47:52.910211+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:47:52.910435+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:47:52.911137+0900 | compress_modules | INFO - Quantizing model.layers.18.self_attn.v_proj using 256 samples
2026-02-10T15:47:53.160612+0900 | compress | METRIC - time 0.25s
2026-02-10T15:47

(20/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:54<00:00,  4.66it/s]

2026-02-10T15:49:00.479965+0900 | compress_modules | INFO - Quantizing model.layers.19.self_attn.q_proj using 256 samples


2026-02-10T15:49:00.862081+0900 | compress | METRIC - time 0.38s
2026-02-10T15:49:00.862711+0900 | compress | METRIC - error 588.82
2026-02-10T15:49:00.863813+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:49:00.864071+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:49:00.865694+0900 | compress_modules | INFO - Quantizing model.layers.19.self_attn.k_proj using 256 samples
2026-02-10T15:49:01.101801+0900 | compress | METRIC - time 0.24s
2026-02-10T15:49:01.102194+0900 | compress | METRIC - error 168.03
2026-02-10T15:49:01.103206+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:49:01.103440+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:49:01.104182+0900 | compress_modules | INFO - Quantizing model.layers.19.self_attn.v_proj using 256 samples
2026-02-10T15:49:01.341678+0900 | compress | METRIC - time 0.24s
2026-02-10T15:49

(21/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:54<00:00,  4.66it/s]

2026-02-10T15:50:08.633553+0900 | compress_modules | INFO - Quantizing model.layers.20.self_attn.q_proj using 256 samples


2026-02-10T15:50:09.020315+0900 | compress | METRIC - time 0.39s
2026-02-10T15:50:09.020805+0900 | compress | METRIC - error 698.36
2026-02-10T15:50:09.021955+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:50:09.022241+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:50:09.023976+0900 | compress_modules | INFO - Quantizing model.layers.20.self_attn.k_proj using 256 samples
2026-02-10T15:50:09.273343+0900 | compress | METRIC - time 0.25s
2026-02-10T15:50:09.273859+0900 | compress | METRIC - error 186.94
2026-02-10T15:50:09.274919+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:50:09.275164+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:50:09.275832+0900 | compress_modules | INFO - Quantizing model.layers.20.self_attn.v_proj using 256 samples
2026-02-10T15:50:09.510996+0900 | compress | METRIC - time 0.23s
2026-02-10T15:50

(22/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:54<00:00,  4.66it/s]

2026-02-10T15:51:16.804300+0900 | compress_modules | INFO - Quantizing model.layers.21.self_attn.q_proj using 256 samples


2026-02-10T15:51:17.188547+0900 | compress | METRIC - time 0.38s
2026-02-10T15:51:17.189046+0900 | compress | METRIC - error 801.29
2026-02-10T15:51:17.190227+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:51:17.190510+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:51:17.192266+0900 | compress_modules | INFO - Quantizing model.layers.21.self_attn.k_proj using 256 samples
2026-02-10T15:51:17.430301+0900 | compress | METRIC - time 0.24s
2026-02-10T15:51:17.430715+0900 | compress | METRIC - error 214.45
2026-02-10T15:51:17.431711+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:51:17.431939+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:51:17.432784+0900 | compress_modules | INFO - Quantizing model.layers.21.self_attn.v_proj using 256 samples
2026-02-10T15:51:17.670362+0900 | compress | METRIC - time 0.24s
2026-02-10T15:51

(23/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:54<00:00,  4.66it/s]

2026-02-10T15:52:24.952356+0900 | compress_modules | INFO - Quantizing model.layers.22.self_attn.q_proj using 256 samples


2026-02-10T15:52:25.342625+0900 | compress | METRIC - time 0.39s
2026-02-10T15:52:25.343169+0900 | compress | METRIC - error 874.18
2026-02-10T15:52:25.344418+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:52:25.344700+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:52:25.346339+0900 | compress_modules | INFO - Quantizing model.layers.22.self_attn.k_proj using 256 samples
2026-02-10T15:52:25.587459+0900 | compress | METRIC - time 0.24s
2026-02-10T15:52:25.587888+0900 | compress | METRIC - error 247.48
2026-02-10T15:52:25.588906+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:52:25.589164+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:52:25.589993+0900 | compress_modules | INFO - Quantizing model.layers.22.self_attn.v_proj using 256 samples
2026-02-10T15:52:25.904096+0900 | compress | METRIC - time 0.31s
2026-02-10T15:52

(24/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:54<00:00,  4.67it/s]

2026-02-10T15:53:33.072875+0900 | compress_modules | INFO - Quantizing model.layers.23.self_attn.q_proj using 256 samples


2026-02-10T15:53:33.466588+0900 | compress | METRIC - time 0.39s
2026-02-10T15:53:33.467036+0900 | compress | METRIC - error 973.81
2026-02-10T15:53:33.468137+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:53:33.468400+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:53:33.470030+0900 | compress_modules | INFO - Quantizing model.layers.23.self_attn.k_proj using 256 samples
2026-02-10T15:53:33.717558+0900 | compress | METRIC - time 0.25s
2026-02-10T15:53:33.717948+0900 | compress | METRIC - error 286.94
2026-02-10T15:53:33.718956+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:53:33.719212+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:53:33.719978+0900 | compress_modules | INFO - Quantizing model.layers.23.self_attn.v_proj using 256 samples
2026-02-10T15:53:33.951926+0900 | compress | METRIC - time 0.23s
2026-02-10T15:53

(25/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:54<00:00,  4.66it/s]

2026-02-10T15:54:41.412883+0900 | compress_modules | INFO - Quantizing model.layers.24.self_attn.q_proj using 256 samples


2026-02-10T15:54:41.809090+0900 | compress | METRIC - time 0.40s
2026-02-10T15:54:41.809583+0900 | compress | METRIC - error 1390.97
2026-02-10T15:54:41.810740+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:54:41.811020+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:54:41.812691+0900 | compress_modules | INFO - Quantizing model.layers.24.self_attn.k_proj using 256 samples
2026-02-10T15:54:42.052354+0900 | compress | METRIC - time 0.24s
2026-02-10T15:54:42.052869+0900 | compress | METRIC - error 369.70
2026-02-10T15:54:42.053833+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:54:42.054113+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:54:42.054906+0900 | compress_modules | INFO - Quantizing model.layers.24.self_attn.v_proj using 256 samples
2026-02-10T15:54:42.292240+0900 | compress | METRIC - time 0.24s
2026-02-10T15:5

(26/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:54<00:00,  4.67it/s]

2026-02-10T15:55:49.459190+0900 | compress_modules | INFO - Quantizing model.layers.25.self_attn.q_proj using 256 samples


2026-02-10T15:55:49.840996+0900 | compress | METRIC - time 0.38s
2026-02-10T15:55:49.841579+0900 | compress | METRIC - error 1585.70
2026-02-10T15:55:49.842640+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:55:49.842904+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:55:49.844501+0900 | compress_modules | INFO - Quantizing model.layers.25.self_attn.k_proj using 256 samples
2026-02-10T15:55:50.081970+0900 | compress | METRIC - time 0.24s
2026-02-10T15:55:50.082366+0900 | compress | METRIC - error 400.92
2026-02-10T15:55:50.083403+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:55:50.083631+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:55:50.084349+0900 | compress_modules | INFO - Quantizing model.layers.25.self_attn.v_proj using 256 samples
2026-02-10T15:55:50.321549+0900 | compress | METRIC - time 0.24s
2026-02-10T15:5

(27/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:54<00:00,  4.66it/s]

2026-02-10T15:56:57.602439+0900 | compress_modules | INFO - Quantizing model.layers.26.self_attn.q_proj using 256 samples


2026-02-10T15:56:57.984254+0900 | compress | METRIC - time 0.38s
2026-02-10T15:56:57.984846+0900 | compress | METRIC - error 1896.53
2026-02-10T15:56:57.985981+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:56:57.986258+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:56:57.987842+0900 | compress_modules | INFO - Quantizing model.layers.26.self_attn.k_proj using 256 samples
2026-02-10T15:56:58.226054+0900 | compress | METRIC - time 0.24s
2026-02-10T15:56:58.226589+0900 | compress | METRIC - error 513.17
2026-02-10T15:56:58.227576+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:56:58.227832+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:56:58.228675+0900 | compress_modules | INFO - Quantizing model.layers.26.self_attn.v_proj using 256 samples
2026-02-10T15:56:58.463978+0900 | compress | METRIC - time 0.24s
2026-02-10T15:5

(28/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:54<00:00,  4.67it/s]

2026-02-10T15:58:05.626424+0900 | compress_modules | INFO - Quantizing model.layers.27.self_attn.q_proj using 256 samples


2026-02-10T15:58:06.046285+0900 | compress | METRIC - time 0.42s
2026-02-10T15:58:06.046720+0900 | compress | METRIC - error 2861.87
2026-02-10T15:58:06.047757+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:58:06.048024+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:58:06.049613+0900 | compress_modules | INFO - Quantizing model.layers.27.self_attn.k_proj using 256 samples
2026-02-10T15:58:06.287871+0900 | compress | METRIC - time 0.24s
2026-02-10T15:58:06.288303+0900 | compress | METRIC - error 736.93
2026-02-10T15:58:06.289301+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:58:06.289561+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:58:06.290307+0900 | compress_modules | INFO - Quantizing model.layers.27.self_attn.v_proj using 256 samples
2026-02-10T15:58:06.525664+0900 | compress | METRIC - time 0.24s
2026-02-10T15:5

(29/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:55<00:00,  4.64it/s]

2026-02-10T15:59:14.118033+0900 | compress_modules | INFO - Quantizing model.layers.28.self_attn.q_proj using 256 samples


2026-02-10T15:59:14.503373+0900 | compress | METRIC - time 0.38s
2026-02-10T15:59:14.503901+0900 | compress | METRIC - error 3286.65
2026-02-10T15:59:14.505067+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:59:14.505335+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:59:14.507064+0900 | compress_modules | INFO - Quantizing model.layers.28.self_attn.k_proj using 256 samples
2026-02-10T15:59:14.759628+0900 | compress | METRIC - time 0.25s
2026-02-10T15:59:14.760044+0900 | compress | METRIC - error 846.64
2026-02-10T15:59:14.761114+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:59:14.761354+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:59:14.762149+0900 | compress_modules | INFO - Quantizing model.layers.28.self_attn.v_proj using 256 samples
2026-02-10T15:59:14.999365+0900 | compress | METRIC - time 0.24s
2026-02-10T15:5

(30/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:56<00:00,  4.54it/s]

2026-02-10T16:00:23.727153+0900 | compress_modules | INFO - Quantizing model.layers.29.self_attn.q_proj using 256 samples


2026-02-10T16:00:24.127465+0900 | compress | METRIC - time 0.40s
2026-02-10T16:00:24.128040+0900 | compress | METRIC - error 3258.94
2026-02-10T16:00:24.129238+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T16:00:24.129566+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T16:00:24.131837+0900 | compress_modules | INFO - Quantizing model.layers.29.self_attn.k_proj using 256 samples
2026-02-10T16:00:24.380610+0900 | compress | METRIC - time 0.25s
2026-02-10T16:00:24.381103+0900 | compress | METRIC - error 923.58
2026-02-10T16:00:24.382247+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T16:00:24.382539+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T16:00:24.383293+0900 | compress_modules | INFO - Quantizing model.layers.29.self_attn.v_proj using 256 samples
2026-02-10T16:00:24.624586+0900 | compress | METRIC - time 0.24s
2026-02-10T16:0

(31/31): Propagating: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:00<00:00, 3975.35it/s]


2026-02-10T16:00:37.454456+0900 | finalize | INFO - Compression lifecycle finalized for 1 modifiers
2026-02-10T16:00:37.460874+0900 | post_process | WARNING - Optimized model is not saved. To save, please provide`output_dir` as input arg.Ex. `oneshot(..., output_dir=...)`

[INFO] 양자화 완료!


# 6. 모델 저장

In [6]:
print("[INFO] 모델 저장 중...")

if os.path.exists(OUT_DIR):
    shutil.rmtree(OUT_DIR)
os.makedirs(OUT_DIR, exist_ok=True)

model.save_pretrained(OUT_DIR, save_compressed=True)
tokenizer.save_pretrained(OUT_DIR)

# 파일 확인
print(f"\n[INFO] 저장된 파일:")
total_size = 0
for f in sorted(os.listdir(OUT_DIR)):
    size = os.path.getsize(os.path.join(OUT_DIR, f))
    total_size += size
    print(f"  {f}: {size/1e6:.1f} MB")

quantized_size_gb = total_size / 1e9

print("\n" + "=" * 60)
print("모델 크기 비교")
print("=" * 60)
print(f"  원본: {ORIGINAL_MODEL_SIZE_GB:.2f} GB")
print(f"  압축: {quantized_size_gb:.2f} GB")
print(f"  압축률: {quantized_size_gb / ORIGINAL_MODEL_SIZE_GB * 100:.1f}%")
print("=" * 60)

[INFO] 모델 저장 중...
2026-02-10T16:00:37.862422+0900 | get_model_compressor | INFO - skip_sparsity_compression_stats set to True. Skipping sparsity compression statistic calculations. No sparsity compressor will be applied.


Compressing model: 210it [00:02, 102.41it/s]



[INFO] 저장된 파일:
  chat_template.jinja: 0.0 MB
  config.json: 0.0 MB
  generation_config.json: 0.0 MB
  merges.txt: 1.2 MB
  model.safetensors: 1407.7 MB
  recipe.yaml: 0.0 MB
  special_tokens_map.json: 0.0 MB
  tokenizer.json: 7.9 MB
  tokenizer_config.json: 0.1 MB
  vocab.json: 1.9 MB

모델 크기 비교
  원본: 2.56 GB
  압축: 1.42 GB
  압축률: 55.4%


# 7. 제출 파일 생성

In [7]:
zip_name = "submit_awq_optimized"
print(f"[INFO] {zip_name}.zip 생성 중...")

if os.path.exists(f"{zip_name}.zip"):
    os.remove(f"{zip_name}.zip")

shutil.make_archive(
    base_name=zip_name,
    format="zip",
    root_dir=".",
    base_dir=OUT_DIR,
)

zip_size = os.path.getsize(f"{zip_name}.zip") / 1e9
print(f"[INFO] 생성 완료: {zip_name}.zip ({zip_size:.2f} GB)")

if zip_size <= 10:
    print("✅ 용량 제한 충족")
else:
    print("❌ 용량 초과!")

[INFO] submit_awq_optimized.zip 생성 중...
[INFO] 생성 완료: submit_awq_optimized.zip (0.88 GB)
✅ 용량 제한 충족


---

# 예상 성능

## Marlin-AWQ 효과

| 지표 | 베이스라인 | AWQ 최적화 |
|------|-----------|------------|
| PerfNorm | ~0.95 | ~0.95 |
| SpeedNorm | ~0.30 | **~0.40** |
| **Score** | ~0.625 | **~0.675** |

## 참고
- [vLLM Quantization Benchmarks](https://docs.jarvislabs.ai/blog/vllm-quantization-complete-guide-benchmarks)
- Marlin-AWQ: 741 tok/s (가장 빠름)

---